src/components/auth/login-form/LoginForm.tsx

```diff
@@ -0,0 +1,116 @@
+import './LoginForm.scss';
+
+import { LoginRequest } from '../../../types/auth/login-request.type';
+import { PRODUCTS_URL_PREFIX } from '../../../constants/api.constants';
+import React from 'react';
+import { setAxiosAccessToken } from '../../../utils/request.utils';
+import { useAuthContext } from '../../../context/AuthContext';
+import { useForm } from 'react-hook-form';
+import { useLogin } from '../../../hooks/auth/useLogin';
+import { useNavigate } from 'react-router-dom';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+
+const loginSchema = z.object({
```

Comment:
**suggestion:** Try to separate form definitions from components. Infer the type using `z.infer` 
Example:
```typescript
export const ChatPayloadApiDtoSchema = z.object({
    id: z.string().optional(),
    assistantId: z.string(),
    userId: z.string(),
    message: z.string(),
    sender: z.string().optional(),
    createdAt: z.string().optional()
});

export type ChatPayloadApiDto = z.infer<typeof ChatPayloadApiDtoSchema>;
```

Reply:
This separation of form schema and types is applicable to all other form components